### PR TITLE
✨ webserver API: generate invitation

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

API to create invitations for PO-role users only
image

- ✨ `services/webserver api version: 0.30.0 → 0.31.0`. 
    - ✨ PRODUCT_OWNER can `generate_invitation`: `POST /v0/invitation:generate` 
    - SEE [swagger openapi.json](https://editor.swagger.io/?url=https://raw.githubusercontent.com/ITISFoundation/osparc-simcore/5760e84225712b9fd9aba9135ba7f0a16ed24e53/services/web/server/src/simcore_service_webserver/api/v0/openapi.yaml)
- ♻️ `services/invitations version: 1.0.1 → 1.0.2`
   - moved models to `models_library` 
   - passed ruff
  - SEE [swagger openapi](https://editor.swagger.io/?url=https://raw.githubusercontent.com/ITISFoundation/osparc-simcore/5760e84225712b9fd9aba9135ba7f0a16ed24e53/services/invitations/openapi.json)

## Related issue/s

- Part of https://github.com/ITISFoundation/osparc-simcore/issues/4752

## How to test

- Driving test 
```cmd
cd services/web/server
pytest -vv tests/unit/with_dbs/03/invitations/test_invitations.py:test_product_owner_generate_invitation
```
- all tests in invitations
```cmd
cd services/invitations
make install-dev
make tests-dev-unit
```

## DevOps 

None

### CODE REVIEW

**Review by matusdrobuliak66 — APPROVED** [review]
Nice, thanks! :+1:

**Review by odeimaiz — APPROVED** [review]
Great!

File: services/web/server/src/simcore_service_webserver/products/_invitations_handlers.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+@routes.post(f"/{VTAG}/invitation:generate", name="generate_invitation")
+@login_required
+@permission_required("product.invitations")
+async def generate_invitation(request: web.Request):
+    req_ctx = _ProductsRequestContext.parse_obj(request)
+    body = await parse_request_body_as(GenerateInvitation, request)
+
+    _, user_email = await get_user_name_and_email(request.app, user_id=req_ctx.user_id)
+
+    # NOTE: check if invitations are activated in this product or raise
```
Comment (GitHK) [inline services/web/server/src/simcore_service_webserver/products/_invitations_handlers.py:43]: is this a disguised todo?

**Review by GitHK — APPROVED** [review]

File: services/web/server/src/simcore_service_webserver/products/_invitations_handlers.py (hunk 1 above)
Comment (pcrespov) [reply services/web/server/src/simcore_service_webserver/products/_invitations_handlers.py:43]: guilty as charged!  It is something i need to check with @odeimaiz ! But it is not relevant!

**Pull request merged**